### PR TITLE
Optimize dependencies handling

### PR DESCRIPTION
**Pull request by ariskotsomitopoulos**
- Fix minSdk version to be aligned to all modules
- Fix & upgrade markwon dependency
- Create a file dependencies.gradle that will contain all the shared & common dependencies between modules in order to avoid inconsistencies

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,22 +1,23 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
-    // Ref: https://kotlinlang.org/releases.html
```
Comment (bmarty) [inline build.gradle:4]: Can you move this comment to the file with all the dependencies please?

File: attachment-viewer/build.gradle [hunk 2]
```diff
@@ -49,12 +50,12 @@ dependencies {
...
+    implementation libs.rx.rxAndroid
 
-    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib:${versions.kotlin}"
     implementation 'androidx.core:core-ktx:1.6.0'
-    implementation 'androidx.appcompat:appcompat:1.3.1'
+    implementation libs.androidx.appCompat
     implementation "androidx.recyclerview:recyclerview:1.2.1"
 
     implementation 'com.google.android.material:material:1.4.0'
```
Comment (bmarty) [inline attachment-viewer/build.gradle:61]: this lib is also on several modules

File: dependencies.gradle [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        'dagger'           : '2.38.1',
+        'timber'           : '5.0.1',
+        'work'             : '2.5.0',
+        'lifecycle'        : '2.2.0',
+        'moshi'            : '1.12.0',
+        'arch'             : '2.1.0',
+        'markwon'          : '4.6.2',
+        'arrow'            : '0.8.2',
+        'rxKotlin'         : '2.4.0',
+        'rxAndroid'        : '2.1.1',
```
Comment (bmarty) [inline dependencies.gradle:20]: Will dependabot be able to understand and update those values?

File: vector/build.gradle [hunk 4]
```diff
@@ -315,16 +318,9 @@ dependencies {
...
     def big_image_viewer_version = '1.8.1'
     def glide_version = '4.12.0'
-    def moshi_version = '1.12.0'
-    def daggerVersion = '2.38.1'
     def autofill_version = "1.1.0"
-    def work_version = '2.5.0'
-    def arch_version = '2.1.0'
-    def lifecycle_version = '2.2.0'
     def rxbinding_version = '3.1.0'
     def jjwt_version = '0.11.2'
```
Comment (bmarty) [inline vector/build.gradle:325]: Why not moving all the `def` values?

**Review by bmarty — COMMENTED** [review]
Thanks, some remarks.
Also owe will have to be careful when exporting the SDK, but I will take care of it.

File: attachment-viewer/build.gradle (hunk 2 above)
Comment (ariskotsomitopoulos) [reply attachment-viewer/build.gradle:61]: Yes, there are a lot of dependencies that should be moved there, I just moved the basic ones for feedback. I will continue with the others.

File: vector/build.gradle (hunk 4 above)
Comment (ariskotsomitopoulos) [reply vector/build.gradle:325]: Sure, it will be done in the next commit

File: dependencies.gradle (hunk 3 above)
Comment (ariskotsomitopoulos) [reply dependencies.gradle:20]: Yes, it should work while it is the same as setting an **ext {}**  in root _build.gradle_

File: dependencies.gradle [hunk 5]
```diff
@@ -5,45 +5,123 @@ ext.versions = [
...
-        'timber'           : '5.0.1',
-        'work'             : '2.5.0',
-        'lifecycle'        : '2.2.0',
-        'moshi'            : '1.12.0',
-        'arch'             : '2.1.0',
-        'markwon'          : '4.6.2',
-        'arrow'            : '0.8.2',
-        'rxKotlin'         : '2.4.0',
-        'rxAndroid'        : '2.1.1',
+// Ref: https://kotlinlang.org/releases.html
```
Comment (bmarty) [inline dependencies.gradle:10]: This ref is use to ensure kotlin and kotlinCoroutines version are aligned, so can you move to the line below please?

Comment (bmarty) [reply dependencies.gradle:10]: We will do that later, let see what happen for dependabot current PRs when this one will be merged

**Review by bmarty — APPROVED** [review]
Thanks for the cleanup!

**Pull request merged**